### ISSUE TITLE
Provide Warning dialogue box before allowing Withdrawn status in Projects

### ISSUE THREAD

**Issue by dandahle** [dandahle/Catalyst-AppVetting#230]
Tested on Edna Finnesgard. I changed status to Withdrawn and then could not change it to In Progress after. The "Complete" status change has a pop up warning, which is a helpful warning for permanent changes. 

Or, don't make Withdrawn a permanent change and allow changing status back to Upcoming or In Progress.

**Comment by rarafon** [comment]
Added a message in commit.

I can also make the reversal of the declined assessments or withdrawn / completed projects possible as well. Right now, it's permanent since everything is set to a complete status which makes prevents editing, but doing the reversal shouldn't be too difficult to implement.

**Comment by dandahle** [comment]
Let me check out your changes and talk to Shawn & team about ability to reverse. I think once projects are "complete" they should never change (this is where your checks are really useful), but sometimes we do have a decision to decline or a client reinstates their application after withdrawal, so reversal may be helpful. But, wait for me to confirm before any further changes. If this works, I'll close the issue.

**Comment by dandahle** [comment]
I checked with Shawn and I'm including his full answer here along with a summary for action after: 

“Generally speaking, I really like the idea of “locking” applications that are withdrawn/declined and projects that are completed, because then those records become a permanent reflection of what actually happened.  However, I because we’re human and because there are multiple people working on the same files, I think there will be lots of times that we’ll want to update some information about a completed project after the fact….for example, at the end of the year when I’m doing my reporting, there’s a huge likelihood that when I run a report for the whole year, the results it spits out won’t be perfect on the first try, and we’ll need to go back into individual records to add things that were forgotten, etc.  I also think that there will be times when we want to re-open a declined or withdrawn application, and if we can’t, that will be a total drag to have to redo everything from scratch. 

So after writing this and processing myself through the process of writing this, I guess what I’m saying is that I’d like the ability to mess with things once they’re completed.”

In short, our documentation never really seems to be complete and we have to go back after the fact and update / change / correct / add things, so locking the data sounds like a good idea to prevent it getting messed up, but in reality would result in less complete and less accurate client and project info for Catalyst. 

If you can figure out a way to provide a warning dialogue box for when those applications and projects are opened to be careful that any changes will alter the project info and be saved, that would be a helpful reminder.

**Comment by rarafon** [comment]
Assessments, projects, work items, and material items can now be edited and are no longer locked.
